src/suffer.cpp

```diff
@@ -857,85 +862,204 @@ std::map<bodypart_id, float> Character::bodypart_exposure()
+        // Sunburn effects occur about 3 times per minute
+        // albinism/datura occur about once per minute
         return;
     }
 
     // TODO: Could factor bodypart_exposure out of Character too
     std::map<bodypart_id, float> bp_exposure = you.bodypart_exposure();
 
-    // Minimum exposure threshold for pain
-    const float MIN_EXPOSURE = 0.01f;
+    enum Sunburn { None, Focus_Loss, Pain, Damage };
+
+    auto heavy_sunburn = [severe, &you]( bodypart_id bp ) {
+        if( severe ) {
+            // Because hands and feet share an HP pool with arms and legs, and the mouth shares
+            // an HP pool with the head, those parts take an unfair share of damage in relation
+            // to the torso, which only has one part.  Increase torso damage to balance this.
+            if( bp == bodypart_id( "torso" ) ) {
+                you.apply_damage( nullptr, bp, 2 );
+            } else {
+                you.apply_damage( nullptr, bp, 1 );
+            }
+            return Damage;
+        } else {
+            you.mod_pain( 1 );
+            return Pain;
+        }
+    };
+    auto medium_sunburn = [severe, &you] {
+        if( severe )
+        {
+            you.mod_pain( 1 );
+            return Pain;
+        } else
+        {
+            you.mod_focus( -1 );
+            return Focus_Loss;
+        }
+    };
+    auto light_sunburn = [severe, &you] {
+        if( severe )
+        {
+            you.mod_focus( -1 );
+            return Focus_Loss;
+        } else
+        {
+            return None;
+        }
+    };
+
     // Track body parts above the threshold
-    std::vector<std::pair<float, bodypart_id>> affected_bodyparts;
-    // Check each bodypart with exposure above the minimum
-    for( const std::pair<const bodypart_id, float> &bp_exp : bp_exposure ) {
-        const float exposure = bp_exp.second;
-        // Skip minimally-exposed parts, and skip the eyes (handled by sunglasses)
-        if( exposure <= MIN_EXPOSURE || bp_exp.first == bodypart_id( "eyes" ) ) {
+    std::map<bodypart_id, Sunburn> affected_bodyparts;
+
+    for( auto &bp_exp : bp_exposure ) {
+        bodypart_id bp = bp_exp.first;
+        float exposure = bp_exp.second;
+
+        if( bp == bodypart_id( "eyes" ) ) {
+            // Sunglasses can keep the sun off the eyes.
+            if( you.has_flag( json_flag_GLARE_RESIST )
+                || you.worn_with_flag( flag_SUN_GLASSES )
+                || you.worn_with_flag( flag_BLIND ) ) {
+                continue;
+            }
+            // If no UV-/glare-protection gear is worn the eyes should be treated as unprotected
+            exposure = 100;
+        } else if( you.get_wielded_item().has_flag( flag_RAIN_PROTECT )
+                   || ( ( bp == body_part_hand_l || bp == body_part_hand_r )
+                        && you.worn_with_flag( flag_POCKETS )
+                        && you.get_wielded_item().volume() < 500_ml )
```

Comment:
That was basically copied from [`character.cpp:bonus_item_warmth`](https://github.com/CleverRaven/Cataclysm-DDA/blob/2b38de5131b4d7abbe1320d2d2a48cb06fe399d7/src/character.cpp#L8075). It could perhaps be extracted into a separate function like `can_use_pocket`. Same for the hood.

Reply:
Have just done that